cli/src/merge_tools/external.rs

```diff
@@ -277,6 +278,13 @@ fn run_mergetool_external_single_file(
     }
 
     let new_file_ids = if editor.merge_tool_edits_conflict_markers || exit_status_implies_conflict {
+        tracing::info!(
+            "jj is reparsing output for conflicts{}.",
+            editor
+                .merge_tool_edits_conflict_markers
+                .then_some(" due to `merge-tool-edits-conflict-markers` config")
+                .unwrap_or_default()
+        );
```

Comment:
nit: just print these boolean variables? That should work so long as the variable names are descriptive.

Maybe I don't fully understand the reason why you want to see "due to merge-tool-edits-conflict-markers config" or not. If we want to check typos in config, I would insert `tracing::debug!(?editor)` or something.

Reply:
I want to print something useful to a user who doesn't understand the internals. For example, they might not know that the underscores in `merge_tool_edits_conflict_markers` correspond to dashes in the option name, or that this even refers to an option. This gives them something to search the docs for.

Reply:
Okay, I misunderstood that this would be for people who are struggling with the tool configuration (and wondered why it is printed only when conflict is implied.)

Maybe something like this will work? (I feel it's a bit simpler and still readable.)
```
tracing::info!(
    "reparsing output for conflicts, merge-tool-edits-conflict-markers = {:?}, \
    exit_status_implies_conflict = {:?}",
    ..
)?;
```

Reply:
OK. I still want to clarify that `exist_status_implied_conflict` is an internal flag while `merge-tool-edits-conflict-markers` is a config one can set.